File: benchmarks/OCaml/safety/simple/sum.ml
let rec sum x = if x <= 0 then 0 else x + sum (x - 1)

[@@@assert "typeof(sum) <: (x:{ x:int | true }) -> { ret : int | ret > x }"]
[@@@assert "typeof(sum) <: (x:{ x:int | x > 1 }) -> { ret : int | ret > x }"]
[@@@assert "typeof(sum) <: (x:{ x:int | true }) -> { ret : int | ret >= x }"]

File: benchmarks/C/safety/sum.c
int sum(int x) {
  if (x <= 0) {
    return 0;
  } else {
    return x + sum(x - 1);
  }
}
main() {
  int x;
  if (sum(x) >= x) {
    return;
  } else {
  ERROR:
    return;
  }
}
